### PR TITLE
Themeable UI

### PR DESCRIPTION
**Pull request by taulinger**
Support for a light and a dark theme

### CODE REVIEW

File: src/main.css [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
-    --bg-color: #212f3d;
-    --color: white;
-    --input-bg-color: #354453;
     --font-family: "Courier New", Courier, monospace;
 }
-body {
+
+.light-theme {
+    --primary-bg-color: white;
+    --secondary-bg-color: #F8F8F8;
```
Comment (thiloSchlemmer) [inline src/main.css:9]: on my display is the colorstep( Hue) between white an  #F8F8F8 to small. The background for the autocompletion-field is visible but the background in the toggle-button for the UI-Theme is not visible (on my screen). I would prever #e8e8e8, which is still noticeable

Comment (taulinger) [reply src/main.css:9]: thx, i will  borrow that. 

The main focus of this PR is not to finish the layout, but to provide a basic infrastructure, upon which
our specialist can do his work (-;

File: test/modules/uiTheme/components/toggle/ThemeToggle.test.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+	describe('when initialized', () => {
+		it('renders the view', async () => {
+			const element = await setup();
+
+			expect(element.shadowRoot.querySelector('.switch').title).toBe('uiTheme_toggle_tooltip_dark');
+			expect(element.shadowRoot.querySelector('input').checked).toBeTrue();
+			expect(element.shadowRoot.querySelector('.icon.adjust')).toBeTruthy();
+			expect(element.shadowRoot.querySelector('.slider.round')).toBeTruthy();
+		});
+	});
```
Comment (thiloSchlemmer) [inline test/modules/uiTheme/components/toggle/ThemeToggle.test.js:38]: all tests here in a wellformed AAA-Manner /arrange/act/assert. 

Do we give tests the structure-comments 
`// arrange
// act 
// assert`
?

Comment (taulinger) [reply test/modules/uiTheme/components/toggle/ThemeToggle.test.js:38]: Not when it's clear what's happening:
https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80

File: src/modules/uiTheme/components/provider/ThemeProvider.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		this._environmentService.getWindow().matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
+			changeTheme(e.matches ? 'dark' : 'light');
+		});
+	}
+
+	_updateCss() {
+		
+		const { theme } = this._state;
+		const cssClassToAdd = theme === 'dark' ? 'dark-theme' : 'light-theme';
+		const cssClassToRemove = theme === 'light' ? 'dark-theme' : 'light-theme';
```
Comment (EnBakir) [inline src/modules/uiTheme/components/provider/ThemeProvider.js:33]: Maybe a bit better readable: "const cssClassToAdd = (theme === 'dark') ? 'dark-theme' : 'light-theme';"

**Review by EnBakir — COMMENTED** [review]
Code was reviewed. No objections.

File: src/modules/uiTheme/components/provider/ThemeProvider.js (hunk 3 above)
Comment (taulinger) [reply src/modules/uiTheme/components/provider/ThemeProvider.js:33]: We should let the linter ensure this. Configuration will follow in a separate PR...

**Pull request merged**